### PR TITLE
(Datadog) Add OTEL instrumentation

### PR DESCRIPTION
**Pull request by JoseRFelix**
## What is the purpose of the change:

Enable tracing in Datadog

### Linear Task

https://linear.app/osmosis/issue/PLAT-323/vercel

## Brief Changelog

- Remove Sentry
- Add tracing in trpc

### CODE REVIEW

**Review by p0mvn — COMMENTED** [review]
How does trace id header propagation to servers work?

For example, wouldn't we need to set headers explicitly when making a request to SQS?

File: packages/web/instrumentation.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+import { getOpentelemetryServiceName } from "~/utils/service-name";
+
+export function register() {
+  registerOTel({
+    serviceName: getOpentelemetryServiceName(),
+    instrumentationConfig: {
+      fetch: {
+        propagateContextUrls: ["*"],
+      },
+    },
```
Comment (JoseRFelix) [inline packages/web/instrumentation.ts:12]: @p0mvn Otel will automatically propagate the context on fetch requests. However, we have to make sure the service is compatible with the header

File: packages/web/utils/error.ts [hunk 2]
```diff
@@ -1,8 +1,6 @@
-import * as Sentry from "@sentry/nextjs";
-
 export function captureError(e: any) {
   if (e instanceof Error) {
-    Sentry.captureException(e);
+    // Sentry.captureException(e);
```
Comment (jonator) [inline packages/web/utils/error.ts:3]: Q: will exceptions be captured some other way?

**Review by jonator — APPROVED** [review]

File: packages/web/utils/error.ts (hunk 2 above)
Comment (JoseRFelix) [reply packages/web/utils/error.ts:3]: Yes, we'll use Opentelemetry. I updated the one on the `server` package but missed this one. Good catch! My latest commit fixes this

**Pull request merged**